drivers/kubevirt/kubevirt.go

```diff
@@ -507,3 +515,330 @@ func AddPVCsToVirtualMachine(vm kubevirtv1.VirtualMachine, pvcs []*corev1.Persis
 
 	return nil
 }
+
+// StartAndWaitForVMIMigration starts the VM migration and waits for the VM to be in running state in the new node
+func StartAndWaitForVMIMigration(virtualMachineCtx *scheduler.Context, ctx context1.Context, schedDriver scheduler.Driver) error {
+
+	vms, err := GetAllVMsFromScheduledContexts([]*scheduler.Context{virtualMachineCtx})
+	if err != nil {
+		return err
+	}
+
+	// Get the namespace and name of the VM
+	if len(vms) == 0 {
+		return fmt.Errorf("no VMs found for VM [%s] in namespace [%s]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace)
+	}
+	vmiNamespace := vms[0].Namespace
+	vmiName := vms[0].Name
+	if len(vms) > 1 {
+		return fmt.Errorf("more than 1 VMs found for VM [%s] in namespace [%s]", virtualMachineCtx.App.Key, virtualMachineCtx.App.NameSpace)
+	}
+
+	//Get the node where the vm is scheduled before the migration
+	nodeName, err := GetNodeOfVM(vms[0])
+	if err != nil {
+		return err
+	}
+	log.Infof("VM [%s] in namespace [%s] is scheduled on node [%s]", vmiName, vmiNamespace, nodeName)
+
+	// Start the VM migration
+	migration, err := kubevirtdy.Instance().CreateVirtualMachineInstanceMigration(ctx, vmiNamespace, vmiName)
+	if err != nil {
+		log.Errorf("Failed to create VM migration for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+		return err
+	}
+	log.Infof("VM migration created for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+
+	// wait for completion
+	var migr *kubevirtdy.VirtualMachineInstanceMigration
+
+	// get volumes from app context
+	volume, err := schedDriver.GetVolumes(virtualMachineCtx)
+	if err != nil {
+		return err
+	}
+
+	t := func() (interface{}, bool, error) {
+		migr, err = kubevirtdy.Instance().GetVirtualMachineInstanceMigration(ctx, vmiNamespace, migration.Name)
+		if err != nil {
+			return "", false, fmt.Errorf("failed to get migration for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+		}
+		if !migr.Completed {
+			return "", true, fmt.Errorf("waiting for migration to complete for VM [%s] in namespace [%s]", vmiName, vmiNamespace)
+		}
+
+		// wait until there is only one pod in the running state
+		testPod, err := getVMPod(virtualMachineCtx, volume[0])
+		if err != nil {
+			return "", true, err
+		}
+
+		//Get the node where the vm is scheduled after the migration
+		nodeNameAfterMigration := testPod.Spec.NodeName
+
+		if nodeName == nodeNameAfterMigration {
+			return "", false, fmt.Errorf("VM pod live migrated [%s] in namespace [%s] but is still on the same node [%s]", testPod.Name, testPod.Namespace, nodeName)
+		}
+		log.InfoD("VM pod live migrated to node: [%s]", nodeNameAfterMigration)
+		return "", false, nil
+	}
+	_, err = task.DoRetryWithTimeout(t, defaultVmMountCheckTimeout, defaultVmMountCheckRetryInterval)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func getVMPod(virtualMachineCtx *scheduler.Context, vol *volume.Volume) (*corev1.Pod, error) {
```

Comment:
getVirtLauncherPodForVM

Reply:
Changed function name getVmpod to `GetVirtLauncherPodForVM`